pkg/controller/updatechannel/updatechannel.go

```diff
@@ -0,0 +1,173 @@
+package updatechannel
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/joho/godotenv"
+	v1 "github.com/rancher/k3os/pkg/apis/k3os.cattle.io/v1"
+	genv1 "github.com/rancher/k3os/pkg/generated/controllers/k3os.cattle.io/v1"
+	"github.com/rancher/k3os/pkg/mode"
+	"github.com/rancher/k3os/pkg/upgrade"
+	"github.com/sirupsen/logrus"
+)
+
+func NewHandler(ctx context.Context, ctl genv1.UpdateChannelController) *handler {
+	env, _ := godotenv.Read("/etc/os-release")
+	return &handler{
+		ctx:   ctx,
+		ctl:   ctl,
+		node:  os.Getenv("K3OS_NODE_NAME"),
+		osrel: env["VERSION_ID"],
```

Comment:
instead of reading from `/etc/os-release` should instead readlink `/k3os/system/k3os/current`.